packages/ddp-server/livedata_server.js

```diff
@@ -1265,13 +1263,15 @@ Object.assign(Subscription.prototype, {
         collectionNames[collectionName] = true;
       };
 
-      this._publishCursorPromise = Promise.all(
-        res.map(c => c._publishCursor(self))
-      )
-        .then(() => {
-          self.ready();
-        })
-        .catch((e) => self.error(e));
+      try {
+        for (const cur of res) {
```

Comment:
why we need to change from parallel handling into sequenced?
`await Promise.all(res.map(cur => cur._publishCursor(self))` seems good.

Reply:
Subscription parameter change needs be handled sync/waited for in the server, otherwise the client presents the behavior reported, being completely removed and then re-added. Not entirely sure how all this works, perhaps some can be handled in parallel but not others.

Reply:
I will try your approach, I think it might still make sense.

Reply:
Looks like it still works, thanks @afrokick !!